src/Radicle/Daemon/MachineStore.hs

```diff
@@ -99,6 +100,11 @@ modifyMachine id f = do
             (m', a) <- f m
             pure (Cached m', a)
 
+modifyMachine' :: forall a m. (MonadMachineStore m) => MachineId -> (Maybe CachedMachine -> m (Maybe CachedMachine, a)) -> m a
+modifyMachine' id f = do
+    machines <- askMachines
+    CMap.modifyValue id (getMachines machines) f
```

Comment:
Would it be possible to not use this function? I'm asking because @jkarni mentioned yesterday that we might want the daemon to stop following machines after a while. I *think* we can implement deleting items in `CMap` if we simplify the API to just `insert`, `delete` and `modifyExisting` which takes as argument a function `a -> m a`, that is, can't delete depending on the value it finds.

Reply:
I think the generalized `modifyValue` is the only way we can prevent these race conditions. Using `insert` and potentially `delete` does not guarantee that request will be linearized.